File: 2.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def addTwoNumbers(self, l1, l2):
        """
        :type l1: ListNode
        :type l2: ListNode
        :rtype: ListNode
        """
        add = 0
        head = tail = None
        while l1 != None or l2 != None or add != 0:
            if l1 != None:
                add += l1.val
                l1 = l1.next
            if l2 != None:
                add += l2.val
                l2 = l2.next
            s = add % 10
            add = add / 10
            if head == None:
                head = tail = ListNode(s)
            else:
                tail.next = ListNode(s)
                tail = tail.next
        return head
            

File: 2.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
public class Solution {
    public ListNode addTwoNumbers(ListNode l1, ListNode l2) {
        ListNode head = null, tail = null;
        int add = 0, s;
        while (l1 != null || l2 != null || add != 0) {
            if (l1 != null) {
                add += l1.val;
                l1 = l1.next;
            }
            if (l2 != null) {
                add += l2.val;
                l2 = l2.next;
            }
            s = add % 10;
            add = add / 10;
            ListNode temp =  new ListNode(s);
            if (head==null) {
                head = tail = temp;
            }
            else {
                tail.next = temp;
                tail = temp;
            }
        }
        return head;
    }
}
